source/sgp_mode/DiversityAnalysis.h

```diff
@@ -0,0 +1,117 @@
+#ifndef DIVERSITY_ANALYSIS_H
+#define DIVERSITY_ANALYSIS_H
+
+#include "../../Empirical/include/emp/Evolve/Systematics.hpp"
+#include "../../Empirical/include/emp/Evolve/World.hpp"
+#include "../../Empirical/include/emp/data/DataFile.hpp"
+#include "../../Empirical/include/emp/math/Random.hpp"
+#include "../../Empirical/include/emp/math/random_utils.hpp"
+#include "../Organism.h"
+#include "../default_mode/Host.h"
+#include "AnalysisTools.h"
+#include "CPU.h"
+#include "CPUState.h"
+#include "GenomeLibrary.h"
+#include "Instructions.h"
+#include "SGPHost.h"
+#include "SGPWorld.h"
+#include "Tasks.h"
+#include "sgpl/algorithm/execute_cpu.hpp"
+#include "sgpl/hardware/Cpu.hpp"
+#include "sgpl/library/OpLibraryCoupler.hpp"
+#include "sgpl/library/prefab/ArithmeticOpLibrary.hpp"
+#include "sgpl/library/prefab/ControlFlowOpLibrary.hpp"
+#include "sgpl/operations/flow_global/Anchor.hpp"
+#include "sgpl/operations/unary/Increment.hpp"
+#include "sgpl/operations/unary/Terminal.hpp"
+#include "sgpl/program/Program.hpp"
+#include "sgpl/spec/Spec.hpp"
+#include "sgpl/utility/ThreadLocalRandom.hpp"
+#include <math.h>
+#include <set>
+
+
+emp::BitSet<64> ReturnTasksDone(CPU org_cpu) {
+  
+  org_cpu.RunCPUStep(emp::WorldPosition::invalid_id, 100);
+  
+
+  return *org_cpu.state.used_resources;
+}
+
+
+
+
+emp::unordered_map<emp::BitSet<64>, int> GetPhenotypeMap(emp::vector<CPU> organisms){ // SGPWorld world
+//pointer??
+    emp::unordered_map<emp::BitSet<64> , int> phenotype;
+    for (size_t i = 0; i < organisms.size(); i++){
+        CPU org_cpu = organisms[i];//.GetCPU();//??
+        emp::BitSet<64> managed_tasks = ReturnTasksDone(org_cpu);
+        
+        // std::cout<<"got phenotype: " << managed_tasks << std::endl; 
+        phenotype[managed_tasks] += 1; // this works with / without key exists (i.e. just like python)
+        
+    }
+    return phenotype;
+
+
+}
+
+
+
+double AlphaDiversityHelper(emp::unordered_map<emp::BitSet<64> , int> phenotype, double q = 0.0) {
+    int pop = 0;
```

Comment:
let's make this a more descriptive variable name since 'pop' is already used to hold the entire population in World and it isn't clear what this is doing